backend-modules/koios/src/main/java/com/bloxbean/cardano/client/backend/koios/KoiosTransactionService.java

```diff
@@ -91,9 +94,15 @@ private TransactionContent convertToTransactionContent(TxInfo txInfo) {
         TransactionContent transactionContent = new TransactionContent();
         transactionContent.setHash(txInfo.getTxHash());
         transactionContent.setBlock(txInfo.getBlockHash());
-        transactionContent.setBlockHeight(txInfo.getBlockHeight());
-        transactionContent.setBlockTime(txInfo.getTxTimestamp());
-        transactionContent.setSlot(txInfo.getAbsoluteSlot());
+        if (txInfo.getBlockHeight() != null) {
```

Comment:
Can we make assumption that any of these value will never be null?  I find my own defensive programming here a bit clunky.

Reply:
@matiwinnetou Thanks for the PR.

These should not be null unless there is a defect on the provider's side. But I think we can keep this null check.

Reply:
ok, feel free to merge when you think this is fine and right time for it.